lib/reports/github_api_client.rb

```diff
@@ -33,19 +35,62 @@ def user_info(username)
       User.new(data["name"], data["location"], data["public_repos"])
     end
 
-    def user_repos(username)
+    def user_repos(username, forks: forks)
       url = "https://api.github.com/users/#{username}/repos"
+      response = client.get(url)
+
+      if response.status == 404
+        raise NonexistentUser, "'#{username}' does not exist"
+      end
+
+      repos = response.body
+
+      link_header = response.headers["link"]
 
+      if link_header
+        while match_data = link_header.match(/<(.*)>; rel="next"/)
+          next_page_url = match_data[1]
+          response = client.get(next_page_url)
+          link_header = response.headers["link"]
+          repos += response.body
+        end
+      end
+
+      repos.map do |repo_data|
+        next if !forks && repo_data["fork"]
+
+        full_name = repo_data["full_name"]
+        language_url = "https://api.github.com/repos/#{full_name}/languages"
+        response = client.get(language_url)
+        Repo.new(repo_data["full_name"], response.body)
+      end.compact
+    end
+
+    def public_events_for_user(username)
```

Comment:
Assignment Branch Condition size for public_events_for_user is too high. [25.38/15]